src/test/java/com/google/firebase/testing/GenericFunction.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.firebase.testing;
+
+/**
+ * A generic operation that can be called with arbitrary arguments.
+ */
+public interface GenericFunction<T> {
```

Comment:
I don't see any code that uses more than one argument. If I remember correctly, this Admin SDK is also moving to Java 7. Can you then just use https://docs.oracle.com/javase/8/docs/api/java/util/function/Function.html?

Reply:
There are some multi arg function calls in FCM tests. But you're right. Once we are on java8 we should be able to clean this up significantly with native `Function` interface and lambdas. Specifically, I plan to replace most anonymous `CallableOp` instances with lambdas.